src/Workspaces/Core/Portable/Shared/Extensions/DocumentExtensions.cs

```diff
@@ -46,29 +46,22 @@ public static bool IsOpen(this Document document)
         /// </summary>
         public static async Task<SemanticModel> GetSemanticModelForSpanAsync(this Document document, TextSpan span, CancellationToken cancellationToken)
         {
-            try
+            var syntaxFactService = document.GetLanguageService<ISyntaxFactsService>();
```

Comment:
don't explicitly crash but let regular our error handling to handle the exception.

Reply:
could you clarify what you mean by this statement?  why would `document.GetLanguageService<ISyntaxFactsService>()` ever crash?

Reply:
I meant try/catch I am deleting.